plugin/semisync/semisync_master.cc

```diff
@@ -1196,6 +1207,51 @@ int ReplSemiSyncMaster::setWaitSlaveCount(unsigned int new_value) {
   return function_exit(kWho, result);
 }
 
+void ReplSemiSyncMaster::signal_semi_sync_ack(const char *const log_file,
+                                              const my_off_t log_pos) {
+  const auto acked = std::make_pair(std::string(log_file), log_pos);
+  mysql_mutex_lock(&LOCK_last_acked);
+  if (acked > last_acked) {
```

Comment:
In theory, this comparison is not 100% correct.
Binlog names like `master-bin.000001` will be increasing to `master-bin.999999` and then will become `master-bin.1000000`
Lexographical comparison of `master-bin.999999` and `master-bin.1000000` does not produce expected results.
Although I admit that having 1 000 000 is not a typical case.
Just wanted to mention this in the comments.